src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -214,9 +216,15 @@ public void openBrowserAndConnect(String serviceProtocolUri) {
       BrowserLauncher.getInstance().browse("http://" + devtoolsHost + ":" + devtoolsPort + "/?hide=debugger&", null);
     }
     else {
-      // TODO: do we need to url encode uri?
+      String urlParam = serviceProtocolUri;
+      try {
+        urlParam = URLEncoder.encode(serviceProtocolUri, "UTF-8");
+      }
+      catch (UnsupportedEncodingException ignored) {
+      }
+
       BrowserLauncher.getInstance().browse(
```

Comment:
Do you want `browse()` to be called even if there is a caught exception in the try block above?  Reading the code here makes it look like `browse()` here is supposed to be inside the try-block.

Reply:
good point; this shouldn't happen - the UTF-8 encoding should be available - but I'll move the browser call up to within the try block in any case